### DISCUSSION TITLE
Additional target types for `absl::SimpleAtoi`?

### DISCUSSION THREAD

**Discussion by 0xERIC** [abseil/abseil-cpp#1753] [General]
Hello, was it intentional that `absl::SimpleAtoi` only works for 32/64/128 bit integers? Would the maintainers be willing to accept a contribution adding support for 8/16 bit integers?

**Comment by derekmauro** [comment]
We generally don't accept these types of contributions (new public functions) from non-Googlers at all. This is because determining if the contribution is safe for our internal codebase is time-consuming and, well, requires access to our internal codebase.

That said, this is currently one of the top feature requests, so someone *may* investigate the possibility soon.

**Reply by 0xERIC** [reply]
Gotcha, thanks for the quick reply!

My impression is that while they'd be declared in a public header, the new functions themselves would be outside of the public API (i.e. inside the `absl::numbers_internal` namespace). But adding them _would_ require modifying `safe_strtoi_base`, which I understand might provoke some concern.

I'll probably patch the functionality in locally and open a PR just because it's only marginally more effort to do so. If the winds change and it can be merged, awesome. If not, that's ok too 🙂 

Thanks again.
